### PR TITLE
- adds support for query parameter attribute

### PR DESCRIPTION
**Pull request by baywet**
related https://github.com/microsoft/kiota/issues/1445

### CODE REVIEW

File: src/QueryParameterAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+namespace Microsoft.Kiota.Abstractions;
+
+/// <summary>
+/// This attribute allows mapping between the query parameter name in the template and the property name in the class.
+/// </summary>
+[AttributeUsage(AttributeTargets.Property | AttributeTargets.Field, Inherited = false, AllowMultiple = false)]
+public sealed class QueryParameterAttribute : Attribute
+{
+    readonly string templateName;
```
Comment (andrueastman) [inline src/QueryParameterAttribute.cs:14]:
```suggestion
    private readonly string templateName;
```

File: Microsoft.Kiota.Abstractions.Tests/QueryParametersBaseTests.cs [hunk 2]
```diff
@@ -13,7 +13,7 @@ public void SetsSelectQueryParameters()
             var requestInfo = new RequestInformation
             {
                 HttpMethod = Method.GET,
-                UrlTemplate = "http://localhost/me?select={select}"
+                UrlTemplate = "http://localhost/me?select={%24select}"
```
Comment (andrueastman) [inline Microsoft.Kiota.Abstractions.Tests/QueryParametersBaseTests.cs:16]:
```suggestion
                UrlTemplate = "http://localhost/me?%24select={select}"
```

**Review by andrueastman — COMMENTED** [review]
Just a couple of comments from me. 
Otherwise we just need to bump version number and release notes in csproj.

File: Microsoft.Kiota.Abstractions.Tests/QueryParametersBaseTests.cs (hunk 2 above)
Comment (baywet) [reply Microsoft.Kiota.Abstractions.Tests/QueryParametersBaseTests.cs:16]:
```suggestion
                UrlTemplate = "http://localhost/me{?%24select}"
```
actually this syntax makes the parameter mandatory, which is not what we want.

**Review by andrueastman — APPROVED** [review]

**Pull request merged**